neptunelib/utils.py

```diff
@@ -31,3 +37,43 @@ def as_list(value):
         return value
     else:
         return [value]
+
+
+def align_channels_on_x(dataframe):
+    channel_dfs, common_x = _split_df_by_stems(dataframe)
+
+    aligned_dfs = []
+    for channel_df in channel_dfs:
+        aligned_df = pd.merge(common_x, channel_df, on='x')
+        aligned_dfs.append(aligned_df)
+
+    aligned_dfs = merge_dataframes(aligned_dfs, on='x', how='outer')
+    return aligned_dfs
+
+
+def get_channel_name_stems(columns):
+    return list(set([col[2:] for col in columns]))
+
+
+def merge_dataframes(dataframes, on, how='outer'):
+    merged_df = functools.reduce(lambda left, right: \
+                                     pd.merge(left, right, on=on, how=how), dataframes)
+    return merged_df
+
+
+def sort_df_by_columns(df):
+    df = df.reindex(sorted(df.columns), axis=1)
+    return df
+
+
+def _split_df_by_stems(df):
+    channel_dfs, x_vals = [], []
+    for stem in get_channel_name_stems(df.columns):
+        channel_df = df[['x_{}'.format(stem), 'y_{}'.format(stem)]]
+        channel_df.columns = ['x', stem]
+        channel_df.dropna(inplace=True)
+        channel_df['x'] = channel_df['x'].astype(int)
```

Comment:
Mhm, ok I can change that. I thought that since we are dropping NaN we wouldn't need it to be float anymore.

Also, the input dataframe (that we get from the former implementation of .get_numerical_values) has some `x_NAME` as int and some `x_NAME` as float (those that have NaN are floats). Should we have all `x_NAME` as float then?

Reply:
Could you point me to where we cast some `x`s to `int`? I don't see it. From what you're describing it seems like a bug.

Reply:
I think pandas.DataFrame infers the type when we do pd.DataFrame(SOMEDATA). And for those columns that don't have NaN and only int-like values it casts them to int. We could assure they stay float by adding `dtype=[float]` to the constructor.

Reply:
I didn't know that. I'd go with the explicit type definition. If the user prefers integers, they can always cast to `int` easily.

Reply:
Sure, on it.